middleware/src/util.go

```diff
@@ -164,3 +167,66 @@ func determineEnableValue(enable rpcmessages.ToggleSetting) string {
 	}
 	return "disable"
 }
+
+// getStringFromRedis returns a string for a given Redis key.
+// If an error occours a ErrorResponse is returned with a ErrorRedisError as error code.
+func (middleware *Middleware) getStringFromRedis(key redis.BaseRedisKey) (value string, errResp rpcmessages.ErrorResponse) {
```

Comment:
this and the next fewfunctions look like they are about converting the error more so than about the actual query.

A good refactor of this would be to have a function `convertRedisError` and only apply it to the error value.